process/block/preprocess/transactions.go

```diff
@@ -149,10 +149,7 @@ func (txs *transactions) IsDataPrepared(requestedTxs int, haveTime func() time.D
 }
 
 // RemoveTxBlockFromPools removes transactions and miniblocks from associated pools
-func (txs *transactions) RemoveTxBlockFromPools(body block.Body, miniBlockPool storage.Cacher) error {
-	if body == nil || body.IsInterfaceNil() {
-		return process.ErrNilTxBlockBody
-	}
+func (txs *transactions) RemoveTxBlockFromPools(body *block.Body, miniBlockPool storage.Cacher) error {
```

Comment:
you are missing body nil check on almost all methods you changed the parameter

Reply:
Done (where looked necessary)